src/Core/LanguageService.fs

```diff
@@ -36,107 +47,150 @@ module LanguageService =
             | LogConfigSetting.DevConsole -> None, true
             | LogConfigSetting.Output -> Some (window.createOutputChannel "F# Language Service"), false
 
-        ConsoleAndOutputChannelLogger(Some "IONIDE-FSAC", INF, channel, logRequestsToConsole)
+        let consoleMinLevel = if logRequestsToConsole then DBG else WRN
+        let inst = ConsoleAndOutputChannelLogger(Some "IONIDE-FSAC", logLanguageServiceRequestsOutputWindowLevel, channel, Some consoleMinLevel)
+        if logLanguageServiceRequestsOutputWindowLevel <> Level.DBG then
+            let levelString = logLanguageServiceRequestsOutputWindowLevel.ToString().Trim()
+            inst.Info ("Logging to output at level %s. If you want detailed messages, try level DBG.", levelString)
+        inst
 
     let genPort () =
         let r = JS.Math.random ()
         let r' = r * (8999. - 8100.) + 8100.
         r'.ToString().Substring(0,4)
 
     let port = genPort ()
-    let private url s = sprintf @"http://localhost:%s/%s" port s
-
+    let private url fsacAction = (sprintf "http://localhost:%s/%s" port fsacAction)
     let mutable private service : child_process_types.ChildProcess option =  None
-
-    let request<'a, 'b> (ep: string) id  (obj : 'a) =
-        log.Debug ("REQUEST  : %s, Request=%j", ep, obj)
-
+    let private platformPathSeparator = if Process.isMono () then "/" else "\\"
+    let private makeRequestId =
+        let mutable requestId = 0
+        fun () -> (requestId <- requestId + 1); requestId
+    let private relativePathForDisplay (path: string) =
+        path.Replace(vscode.workspace.rootPath + platformPathSeparator, "~" + platformPathSeparator)
+    let private makeOutgoingLogPrefix =
+        let outgoingLogFormat = "REQ ({0:000}) ->"
+        fun (id:int) -> String.Format(outgoingLogFormat, id)
+    let private makeIncomingLogPrefix =
+        let incomingLogFormat = "RES ({0:000}) <-"
+        fun (id:int) -> String.Format(incomingLogFormat, id)
+
+    let private logOutgoingRequest id (fsacAction:string) obj =
+        // log.Debug (makeOutgoingLogPrefix(id) + " %s: Request=%j", fsacAction, obj)
         // At the INFO level, it's nice to see only the key data to get an overview of
         // what's happening, without being bombarded with too much detail
-        if JS.isDefined (obj?FileName) then
-            log.Info ("REQUEST  : %s, FileName=%j", ep, obj?FileName)
-        else if JS.isDefined (obj?Symbol) then
-            log.Info ("REQUEST  : %s, Symbol=%j", ep, obj?Symbol)
-        else
-            log.Info ("REQUEST  : %s", ep)
-
-        ax.post (ep, obj)
+        let extraPropInfo =
+            if (JS.isDefined (obj?FileName)) then Some ", File = \"%s\"", Some (relativePathForDisplay (obj?FileName |> unbox))
```

Comment:
`(JS.isDefined (obj?FileName))` ->  `JS.isDefined (obj?FileName)` the instances below too
If you can strip `(obj?FileName)` then those as well